onnxruntime/core/graph/model.cc

```diff
@@ -441,11 +441,11 @@ Status Model::Load(int fd, ONNX_NAMESPACE::ModelProto& model_proto) {
 
 Status Model::Load(int fd, std::shared_ptr<Model>& p_model, const IOnnxRuntimeOpSchemaRegistryList* local_registries,
                    const logging::Logger& logger) {
-  ModelProto model_proto;
+  auto model_proto = onnxruntime::make_unique<ModelProto>();
```

Comment:
allocate on the heap once, make desired changes to the instance, and pass on ownership downstream when it is not needed anymore.